From 24ba7955634dd571a4c34dd712dc8a592eea4d73 Mon Sep 17 00:00:00 2001
From: Joe Harvell <jharvell@dogpad.net>
Date: Mon, 27 Jun 2011 23:20:47 +0200
Subject: [PATCH] Only print the deprecation warning for --chuid/-c when using it

The deprecation warning has been printed when using the replecement functions as
well, bug 373243.
---
 src/rc/start-stop-daemon.c |    5 ++---
 1 files changed, 2 insertions(+), 3 deletions(-)

diff --git a/src/rc/start-stop-daemon.c b/src/rc/start-stop-daemon.c
index b5c2b6e..3017701 100644
--- a/src/rc/start-stop-daemon.c
+++ b/src/rc/start-stop-daemon.c
@@ -788,12 +788,11 @@ start_stop_daemon(int argc, char **argv)
 			background = true;
 			break;
 
-		case 'u':  /* --user <username>|<uid> */
 		case 'c':  /* --chuid <username>|<uid> */
-		{
 			/* DEPRECATED */
 			ewarn("WARNING: -c/--chuid is deprecated and will be removed in the future, please use -u/--user instead");
-
+		case 'u':  /* --user <username>|<uid> */
+		{
 			p = optarg;
 			tmp = strsep(&p, ":");
 			changeuser = xstrdup(tmp);
-- 
1.7.3.4

